src/dvc_data/hashfile/meta.py

```diff
@@ -64,8 +66,14 @@ def from_info(
 
     @classmethod
     def from_dict(cls, d: Dict[str, Any]) -> "Meta":
-        kwargs = {field: d[field] for field in fields_dict(cls) if field in d}
+        kwargs = {}
+        for field in cls.fields:
+            if field in d:
+                kwargs[field] = d[field]
         return cls(**kwargs)
 
     def to_dict(self) -> Dict[str, Any]:
         return asdict(self, recurse=False, filter=_filter_default_or_none)
+
+
+Meta.fields = list(fields_dict(Meta))
```

Comment:
Also trying to save from calling `fields_dict()` every time `from_dict()` is called, we can save it to class attributes once and use it instead.

Reply:
@skshetry Seems like this hack is not worth the effort and perf gain.

Reply:
I also needed this in `dvc` for `Annotation`. But I approached it a bit differently. From that experience, I can say that this is much cleaner. We want to treat that as a ClassVar and this offers us that.

https://github.com/iterative/dvc/blob/f1b666099fa0b83d551e534a74b2d6200b026215/dvc/annotations.py#L27